CMakeLists.txt

```diff
@@ -178,6 +178,23 @@ else()
   set(_private PRIVATE)
 endif()
 
+if(MSVC)
```

Comment:
not too fan of adding those in the CMakeLists, I'd prefer to add them to the CI instead, as it may break downstream builds who don't want to inherit -Werror (e.g. if you use add_subdirectory(libremidi) and then link to the cmake target)

Reply:
I think inheriting the warnings should be prevented by this line ```
target_compile_options(libremidi PUBLIC ${WARNING_FLAGS})
```
a few lines down, since the flags will only apply to the library itself. 
I can still wrap it around an option though if you'd like.

Reply:
Wrapped the warnings around a new option to disable them. I don't think it should be able to leak out since i also changed the target_compile_options to be private but this is just in case.